--- a/Zend/zend_interfaces.h	2022-01-19 08:52:03.000000000 +0900
+++ b/Zend/zend_interfaces.h	2022-01-18 21:16:19.000000000 +0900
@@ -64,6 +75,9 @@
 ZEND_API int zend_user_serialize(zval *object, unsigned char **buffer, size_t *buf_len, zend_serialize_data *data);
 ZEND_API int zend_user_unserialize(zval *object, zend_class_entry *ce, const unsigned char *buf, size_t buf_len, zend_unserialize_data *data);
 
+ZEND_API int zend_class_serialize_deny(zval *object, unsigned char **buffer, size_t *buf_len, zend_serialize_data *data);
+ZEND_API int zend_class_unserialize_deny(zval *object, zend_class_entry *ce, const unsigned char *buf, size_t buf_len, zend_unserialize_data *data);
+
 ZEND_API zend_result zend_create_internal_iterator_zval(zval *return_value, zval *obj);
 
 END_EXTERN_C()
