--- a/libitm/beginend.cc	2021-04-08 20:56:30.157765000 +0900
+++ b/libitm/beginend.cc	2022-01-08 16:46:37.564921635 +0900
@@ -434,7 +434,6 @@
 
   /* FIXME!  Assignment of an aatree like alloc_actions is unsafe; if either
    *this or *tx is destroyed, the other ends up pointing to a freed node.  */
-#pragma GCC diagnostic warning "-Wdeprecated-copy"
   alloc_actions = tx->alloc_actions;
 
   user_actions_size = tx->user_actions.size();
