--- a/src/php_sdl_mixer.h	2022-03-15 22:40:27.000000000 +0900
+++ b/src/php_sdl_mixer.h	2022-03-23 22:48:53.305289000 +0900
@@ -14,15 +14,11 @@
 
 #define PHP_SDL_MIXER_VERSION "0.4.0"
 
-#if defined(__APPLE__) && defined(__MACH__)
 #include <SDL2/SDL.h>
-#else
-#include <SDL.h>
-#endif
 
 #include <php.h>
 #include <ext/standard/info.h>
-#include "SDL_mixer.h"
+#include <SDL2/SDL_mixer.h>
 
 #ifdef  __cplusplus
 } // extern "C"

