--- a/libgcc/gthr-posix.h	2021-07-28 15:55:08.812308000 +0900
+++ b/libgcc/gthr-posix.h	2021-05-14 17:42:10.000000000 +0900
@@ -87,8 +87,7 @@
 #  define __gthrw_pragma(pragma)
 # endif
 # define __gthrw2(name,name2,type) \
-  static __typeof(type) name \
-    __attribute__ ((__weakref__(#name2), __copy__ (type))); \
+  static __typeof(type) name __attribute__ ((__weakref__(#name2))); \
   __gthrw_pragma(weak type)
 # define __gthrw_(name) __gthrw_ ## name
 #else

