--- a/configure	2022-01-22 10:21:51.000000000 +0900
+++ b/configure	2022-01-22 10:33:08.342051622 +0900
@@ -4063,10 +4063,7 @@
 if test "$PHP_INOTIFY" != "no"; then
 
   if test "$cross_compiling" = yes; then :
-  { { $as_echo "$as_me:${as_lineno-$LINENO}: error: in \`$ac_pwd':" >&5
-$as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
-as_fn_error $? "cannot run test program while cross compiling
-See \`config.log' for more details" "$LINENO" 5; }
+  
 else
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */

