--- a/libstdc++-v3/include/bits/char_traits.h	2021-06-01 16:53:06.828501000 +0900
+++ b/libstdc++-v3/include/bits/char_traits.h	2021-12-30 21:40:06.131578305 +0900
@@ -482,15 +482,19 @@
 	    && __constant_char_array_p(__s2, __n))
 	  return __gnu_cxx::char_traits<char_type>::compare(__s1, __s2, __n);
 #endif
-	return wmemcmp(__s1, __s2, __n);
+	if (__n == 0)
+	  return 0;
+	else
+	  return wmemcmp(__s1, __s2, __n);
       }
 
-      static _GLIBCXX17_CONSTEXPR size_t
+      static _GLIBCXX17_CONSTEXPR size_t 
       length(const char_type* __s)
       {
 #if __cplusplus >= 201703L
 	if (__constant_string_p(__s))
 	  return __gnu_cxx::char_traits<char_type>::length(__s);
+	else
 #endif
 	return wcslen(__s);
       }
@@ -506,7 +510,10 @@
 	    && __constant_char_array_p(__s, __n))
 	  return __gnu_cxx::char_traits<char_type>::find(__s, __n, __a);
 #endif
-	return wmemchr(__s, __a, __n);
+	if (__n == 0)
+	  return 0;
+	else
+	  return wmemchr(__s, __a, __n);
       }
 
       static _GLIBCXX20_CONSTEXPR char_type*

